src/App.tsx

```diff
@@ -1,25 +1,57 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
+import { Pagination } from './components/Pagination';
 import { getNumbers } from './utils';
 
-// eslint-disable-next-line @typescript-eslint/no-unused-vars
-const items = getNumbers(1, 42).map(n => `Item ${n}`);
+const getPageItems = (
+  total: number,
+  itemsPerPage: number,
+  currentPage: number,
+) => {
+  return [...getNumbers(1, total)].slice(
+    itemsPerPage * (currentPage - 1),
+    itemsPerPage * currentPage,
+  );
+};
+
+const ITEMS_COUNT = 42;
+const PER_PAGE_COUNT = 5;
 
 export const App: React.FC = () => {
+  const [perPage, setPerPage] = useState(PER_PAGE_COUNT);
+  const [currentPage, setCurrentPage] = useState(1);
+
+  const pageItems = getPageItems(ITEMS_COUNT, perPage, currentPage);
+
+  const handlePerPageSelectorChange = (
+    event: React.ChangeEvent<HTMLSelectElement>,
+  ) => {
+    setPerPage(+event.target.value);
```

Comment:
Converting the event target value to a number using a unary plus (+) is not immediately clear. It's better to use Number(event.target.value) for clarity.